posthog/temporal/data_imports/pipelines/mysql/mysql.py

```diff
@@ -0,0 +1,257 @@
+import dataclasses
+from typing import Any, Optional
+from collections.abc import Iterator
+from django.conf import settings
+import pyarrow as pa
+import pymysql
+from pymysql.cursors import SSCursor, Cursor
+
+from posthog.temporal.common.logger import FilteringBoundLogger
+from posthog.temporal.data_imports.pipelines.helpers import incremental_type_to_initial_value
+from posthog.temporal.data_imports.pipelines.pipeline.typings import SourceResponse
+from posthog.temporal.data_imports.pipelines.pipeline.utils import (
+    DEFAULT_NUMERIC_PRECISION,
+    DEFAULT_NUMERIC_SCALE,
+    build_pyarrow_decimal_type,
+    table_from_iterator,
+)
+from posthog.temporal.data_imports.pipelines.sql_database.settings import DEFAULT_CHUNK_SIZE
+from posthog.warehouse.models import IncrementalFieldType
+
+from dlt.common.normalizers.naming.snake_case import NamingConvention
+
+
+def _sanitize_identifier(identifier: str) -> str:
+    if not identifier.isidentifier() or not identifier.replace(".", "").replace("_", "").isalnum():
```

Comment:
The first part of this conditional means we don't support users with identifiers that are not valid python identifiers. This means we will reject several table names that are valid in mysql but not valid python identifiers. Looking at the [mysql docs](https://dev.mysql.com/doc/refman/8.4/en/identifiers.html), I could find at least two examples:
* Identifiers that start with a number
* Identifiers with a $

Reply:
Also, it is possible to have emojis in table names. This is valid in mysql:
```
create table `ohgod😊why` (i int);
```
But not a valid python identifier (perhaps for a good reason).

Reply:
Nice catch - have catered for these!